### DISCUSSION TITLE
can  BeeGFS utilize multiple network cards simultaneously ？

### DISCUSSION THREAD

**Discussion by yxmingming** [ThinkParQ/beegfs#9] [Q&A] [answered]
Dear Development Team,
     I have configured multiple network cards  in
`    /etc/beegfs/tcp-only-interfaces.conf
`   
     I would like to inquire whether BeeGFS can utilize multiple network cards simultaneously for data transmission and how to configure this.
 According to the configuration file, it appears possible to set up multiple network cards. However, we are currently experiencing a situation where only one network card is being used for file uploads and downloads. Even when the network reaches its maximum bandwidth, the second network card does not share the traffic load. Could you please provide guidance on how to enable multiple network cards to work together to increase the cluster’s throughput bandwidth?

Thank you.

**Comment by iamjoemccormick** [comment, marked as answer]
Hi @yxmingming, 

This is expected behavior. In BeeGFS when you specify multiple network cards/interfaces, only the first one listed is used unless that interface fails, at which point connections will fall back to the next interface (and so on). If you want to use multiple network cards/interfaces simultaneously you should configure network bonding (i.e., link aggregation) in Linux. This will combine multiple physical interfaces into a logical interface that you can then use with BeeGFS.

One exception to this is if you're using RDMA instead of TCP. Then you can configure [BeeGFS clients to automatically use multiple RDMA NICs](https://doc.beegfs.io/latest/advanced_topics/rdma_support.html#client-multi-rail-support). However on the server-side still only a single interface is used unless you configure bonding.

**Reply by dhrvg** [reply]
Hi @iamjoemccormick, to my understanding RoCE V2 doesn't work on LACP bonded interfaces, RDMA bonding is also supported on very limited devices. Is it safe to conclude that for a server to make use of aggregate bonded bandwidth TCP is the only option?

**Reply by iamjoemccormick** [reply]
That is my understanding as well. If you want to use multiple RDMA interfaces on a BeeGFS metadata or storage server you could use BeeGFS [multi mode](https://doc.beegfs.io/latest/advanced_topics/multimode.html) which essentially lets you run one BeeGFS meta/storage service per RDMA interface.
